wandb/sdk/launch/github_reference.py

```diff
@@ -237,7 +237,7 @@ def _update_path(self, dst_dir: str) -> None:
             return
         path = Path(dst_dir, self.path)
         if path.is_file():
-            self.directory = str(path.parent.absolute())
+            self.directory = self.path.replace(path.name, "")
```

Comment:
Note this change, the `path.parent.absolute()` was grabbing the entire tempdir path, but only when the path was a file, not dir. this inconsistency was driving me crazy, now should behave as expected. Might have downstream effects though....

Reply:
Should this potentially risky change be put in a separate PR? Does this PR rely on this change?

Reply:
@KyleGoyette Possibly, but as far as I can tell nothing else is using github_reference anymore. it only affected my test, but in a way that was positive imo.

Reply:
We use this class to fetch Git Repos for our git sourced jobs. I don't think we use the directory there.

Reply:
I can move it, this PR will fail. @KyleGoyette

Reply:
This is fine. I just want to validate that a) this class is used and b) these downstream effects won't break anything critical. It's just concerning to be told this change may have downstream effects, should I investigate them?

I think this change is fine, since the directory attribute seems to make sense.

Reply:
I'm not really sure how it possibly could have been used before because depending on the git link the `ref.directory` was totally different, for example: 

`https://github.com/gtarpenning/wandb-launch-test/test/main.py` --> `ref.directory = /var/folders/s5/drh9pbm55xbbb9mml72qh7xc0000gn/T/tmpmno8xzx5/test/`

but

`https://github.com/gtarpenning/wandb-launch-test/test/` --> `ref.directory = test/`

Reply:
Should it point to the directory containing the code?

Reply:
Currently it points to the relative path to the directory of the reference link. Because you **must** give the gitref.fetch a local directory to fetch to, it feels more useful to leave the gitref.directory a relative path. Then the consumer has a choice to reference files in the remote repo **or** locally.

Reply:
Otherwise to obtain the directory path (for example when constructing an entrypoint) the tempdir would have to be manually removed from the directory path, which is funky

Reply:
okay!